File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/dagger/module/StripeModule.java
package com.lattis.ellipse.presentation.dagger.module;

import android.content.Context;

import com.lattis.ellipse.presentation.dagger.qualifier.DatabaseName;
import com.lattis.ellipse.presentation.dagger.qualifier.StripeKey;
import com.stripe.android.PaymentConfiguration;
import com.stripe.android.Stripe;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.lattis.ellipse.R;

@Module
public class StripeModule {

    @Provides
    @Singleton
    @StripeKey
    String provideStripeKey(Context context) {
        return context.getString(R.string.stripe_key);
    }

    @Provides
    @Singleton
    Stripe provideStripeObject(@StripeKey String stripeKey, Context context) {
        PaymentConfiguration.init(stripeKey);
        return new Stripe(context,
                PaymentConfiguration.getInstance().getPublishableKey());
    }


}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/infrastructure/di/module/StripeModule.kt
package com.lattis.lattis.infrastructure.di.module

import android.content.Context
import com.stripe.android.PaymentConfiguration
import com.stripe.android.Stripe
import dagger.Module
import dagger.Provides
import io.lattis.lattis.R
import javax.inject.Named
import javax.inject.Singleton

@Module
class StripeModule {
    @Provides
    @Singleton
    @Named("StripeKey")
    fun provideStripeKey(context: Context): String {
        return context.getString(R.string.stripe_key)
    }

    @Provides
    @Singleton
    fun provideStripeObject(
        @Named("StripeKey") stripeKey: String,
        context: Context
    ): Stripe {
        PaymentConfiguration.init(context,stripeKey)

        return Stripe(
            context,
            PaymentConfiguration.getInstance(context).publishableKey
        )
    }
}